### PR TITLE
Migrate Client API and Admin API to FCRMessage pkg.

### PR DESCRIPTION
**Pull request by wcgcyx**

### CODE REVIEW

**Review by ssaidali — APPROVED** [review]

**Review by prototypo — COMMENTED** [review]
What happens if the TCP error is anything other than a timeout?

**Comment by wcgcyx** [comment]
> What happens if the TCP error is anything other than a timeout?

I would just drop the connection. The connection is very likely dopped by the peer. 

The timeout error from my understanding means the connection is still live, the peer still responds in TCP layer, but fails to provide any FCR response in a given time. In some cases, this timeout can be ignored (For example when the message is expired, the peer does not have to respond.) and the connection can be kept.

File: internal/api/clientapi/establishment.go [hunk 1]
```diff
@@ -2,48 +2,39 @@ package clientapi
...
 		return
 	}
 
-	response.ProtocolVersion = clientAPIProtocolVersion
-	response.ProtocolSupported = clientAPIProtocolSupported
-	response.GatewayID = g.gateway.GatewayID.ToString()
-
-	sig, err := fcrcrypto.SignMessage(g.gateway.GatewayPrivateKey, g.gateway.GatewayPrivateKeyVersion, *response)
+	// Construct message
+	sig, err := fcrcrypto.SignMessage(g.GatewayPrivateKey, g.GatewayPrivateKeyVersion, challenge)
```
Comment (drinkcoffee) [inline internal/api/clientapi/establishment.go:37]: The signature needs to be across the response message, not just the challenge.

Comment (wcgcyx) [reply internal/api/clientapi/establishment.go:37]: > The signature needs to be across the response message, not just the challenge.

I thought it was just across the challenge. Maybe the engineering plan needs to be updated. I can fix this in the next PR.

**Pull request merged**